Tests failing during compile

While compiling 1e5ae05a3a03c1a94fa16ed9563a45d3e97062d1

```
make  check-TESTS
make[1]: Entering directory '/home/txtsd/build/yaourt-tmp-txtsd/aur-libeatmydata/src/libeatmydata'
make[2]: Entering directory '/home/txtsd/build/yaourt-tmp-txtsd/aur-libeatmydata/src/libeatmydata'
PASS: libeatmydata/test/eatmydatatest_largefile
FAIL: libeatmydata/test/tst-invalidfd
FAIL: libeatmydata/test/eatmydatatest
FAIL: libeatmydata/test/tst-cancel4
FAIL: libeatmydata/test/fcntltest
FAIL: libeatmydata/test/tst-key4
FAIL: libeatmydata/test/fsynctest
============================================================================
Testsuite summary for libeatmydata 128
============================================================================
# TOTAL: 7
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  6
# XPASS: 0
# ERROR: 0
============================================================================
See ./test-suite.log
Please report to http://www.flamingspork.com/projects/libeatmydata
============================================================================
make[2]: *** [Makefile:1130: test-suite.log] Error 1
make[2]: Leaving directory '/home/txtsd/build/yaourt-tmp-txtsd/aur-libeatmydata/src/libeatmydata'
make[1]: *** [Makefile:1238: check-TESTS] Error 2
make[1]: Leaving directory '/home/txtsd/build/yaourt-tmp-txtsd/aur-libeatmydata/src/libeatmydata'
make: *** [Makefile:1484: check-am] Error 2

```

`test-suite.log`
```
========================================
   libeatmydata 128: ./test-suite.log
========================================

# TOTAL: 7
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  6
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: libeatmydata/test/fsynctest
=================================

grep: test.result.run: No such file or directory
rm: cannot remove 'test.result.run': No such file or directory
FAIL libeatmydata/test/fsynctest (exit status: 1)

FAIL: libeatmydata/test/tst-key4
================================

grep: test.result.run: No such file or directory
rm: cannot remove 'test.result.run': No such file or directory
FAIL libeatmydata/test/tst-key4 (exit status: 1)

FAIL: libeatmydata/test/tst-invalidfd
=====================================

grep: test.result.run: No such file or directory
rm: cannot remove 'test.result.run': No such file or directory
FAIL libeatmydata/test/tst-invalidfd (exit status: 1)

FAIL: libeatmydata/test/eatmydatatest
=====================================

grep: test.result.run: No such file or directory
rm: cannot remove 'test.result.run': No such file or directory
FAIL libeatmydata/test/eatmydatatest (exit status: 1)

FAIL: libeatmydata/test/fcntltest
=================================

grep: test.result.run: No such file or directory
rm: cannot remove 'test.result.run': No such file or directory
FAIL libeatmydata/test/fcntltest (exit status: 1)

FAIL: libeatmydata/test/tst-cancel4
===================================

early cancel test of 'fsync' successful
early cancel test of 'fdatasync' successful
early cancel test of 'msync' successful
grep: test.result.run: No such file or directory
rm: cannot remove 'test.result.run': No such file or directory
FAIL libeatmydata/test/tst-cancel4 (exit status: 1)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,7 @@
 language: c
 
+sudo: required
+
 addons:
   apt:
     packages:

--- a/test_run.sh
+++ b/test_run.sh
@@ -57,7 +57,7 @@ if [[ $OSTYPE == *"darwin"* ]]; then
   while [[ "$(ps -o state= -p $test_pid)" != *"T+"* ]]; do
         sleep 1
   done
-  dtruss -p $test_pid 2>> test.result.run &
+  dtruss -p $test_pid 2>> $name-test.result.run &
   dtruss_pid=$!
   kill -CONT $test_pid
   eret=wait $test_pid
@@ -65,7 +65,7 @@ if [[ $OSTYPE == *"darwin"* ]]; then
   wait $dtruss_pid
 else
   export LD_PRELOAD=./.libs/libeatmydata.so
-  strace -o test.result.run "$1" "$LIBEATMYDATA_TEST_ARGS"
+  strace -o $name-test.result.run "$1" "$LIBEATMYDATA_TEST_ARGS"
   eret=$?
 fi
 
@@ -79,9 +79,9 @@ if [[ $OSTYPE == *"darwin"* ]]; then
     SYNC_IN_TRACE="$SYNC_IN_TRACE|^fcntl\\\(.*, 0x33"
 fi
 
-grep "$SYNC_IN_TRACE" test.result.run
+grep "$SYNC_IN_TRACE" $name-test.result.run
 ret=$?
-rm test.result.run
+rm $name-test.result.run
 if [ $ret == 1 ]; then
 	exit 0;
 else

